pkg/url/url.go

```diff
@@ -77,20 +69,24 @@ func List(client *occlient.Client, componentName string, applicationName string)
 	glog.V(4).Infof("Listing routes with label selector: %v", labelSelector)
 	routes, err := client.ListRoutes(labelSelector)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to list route names")
+		return UrlList{}, errors.Wrap(err, "unable to list route names")
 	}
 
-	var urls []URL
+	var urls []Url
 	for _, r := range routes {
-		urls = append(urls, URL{
-			Name:     r.Labels[urlLabels.URLLabel],
-			URL:      r.Spec.Host,
-			Protocol: getProtocol(r),
-			Port:     r.Spec.Port.TargetPort.IntValue(),
-		})
+		a := getMachineReadableFormat(r)
+		urls = append(urls, a)
 	}
 
-	return urls, nil
+	urlList := UrlList{
```

Comment:
we can use `NewURLList()` here to hide the manual setting of `TypeMeta` and `ListMeta`.

Reply:
or `getMachineReadableFormat`

Reply:
:+1: